processor/spanmetricsprocessor/processor.go

```diff
@@ -68,22 +69,31 @@ type processorImp struct {
 	// Additional dimensions to add to metrics.
 	dimensions []Dimension
 
+	// Additional resourceAttributes to add to metrics.
+	resourceAttributes []Dimension
+
 	// The starting time of the data points.
 	startTime time.Time
 
 	// Call & Error counts.
-	callSum map[metricKey]int64
+	callSum map[resourceKey]map[metricKey]int64
 
 	// Latency histogram.
-	latencyCount         map[metricKey]uint64
-	latencySum           map[metricKey]float64
-	latencyBucketCounts  map[metricKey][]uint64
+	latencyCount         map[resourceKey]map[metricKey]uint64
```

Comment:
Why does this logic need to change, and why does the map structure need to become more complex? Currently it works in a way where each unique metric key is mapped to a unique attribute set. Can the metricKey just take into account the resource attributes, rather than having an entirely new key?

Reply:
This enables quicker look up time. The way OTel defines the structure to hold metrics is some thing roughly like (I've omitted the instrumentation library to keep things simplified, if you want to see the actual structure you can check out the metric proto definition):

--> Resource 1
   -> resource attributes (service.name == my.service.1, etc)
   -> metric 1
        -> data point 1
        -> data point 2
  -> metric 2
       -> data point 3
-> Resource 2
    -> resource attributes (service.name == my.service.2, etc)
    -> metric 3
      -> data point 4
etc

This means we need to index on resource key since to build that metric structure we need to find all the metrics/data points that have the same resource attributes and insert that way. (There wasn't any way to lookup on this structure afaict, it's just indexed by integers so we needed to insert all the relevant metrics/dp at the time of creation of the resource/resource attribute structure). 

If we continued to use the singular map structure that they used, we'd need to iterate over the entire metric key set and check if the key is prepended with the same resource attributes for every resource attributes - this would increase the time complexity.